lib/1.4/dml-builtins.dml

```diff
@@ -568,21 +581,76 @@ session dml_qname_cache_t _qname_cache;
-    param writable default this.configuration != "none";
+template _conf_attribute is object {
+    shared method set_attribute(attr_value_t value) -> (set_error_t);
+    shared method get_attribute() -> (attr_value_t);
+
+    param _attr_name : const char *;
+    param _attr_name auto;
+    param _parent_obj_class : conf_class_t **;
+    param _parent_obj_class auto;
+    param _parent_obj_proxy_info : _attr_parent_obj_proxy_info_t;
+    param _parent_obj_proxy_info auto;
+    param _attr_start_dim : uint32;
+    param _attr_start_dim = _object_parent.indices.len;
+    param _attr_allow_cutoff : bool;
+    param _attr_allow_cutoff = (objtype == "connect"
+                                && configuration == "optional");
+
+    param configuration     default "optional";
+    #if (configuration != "required" && configuration != "optional"
+         && configuration != "pseudo" && configuration != "none") {
+        error "'configuration' parameter must be 'required', 'optional', "
+            + "'pseudo', or 'none'";
+    }
+    param _documentation    : const char *;
+    param _documentation    = (   defined(documentation)
+                               #? documentation
+                               #: shown_desc != NULL
+                               #? shown_desc
+                               #: objtype == "register"
+                               #? "register " + _static_qname
+                               #: "Undocumented")
+                            + (   objtype == "connect"
+                               #? defined(_interfaces_doc)
+                               #? _interfaces_doc
+                               #: "" #: "");
+    param _flags            : attr_attr_t;
+    param _flags            = (   configuration == "required"
+                               #? Sim_Attr_Required
+                               #: configuration == "optional"
+                               #? Sim_Attr_Optional
+                               #: configuration == "pseudo"
+                               #? Sim_Attr_Pseudo
+                               #: 0)
+                             | (persistent #? Sim_Attr_Persistent #: 0)
+                             | (internal #? Sim_Attr_Internal #: 0);
+    param _generated        : bool;
+    param _generated        = configuration != "none";
+    param persistent        default false;
+    param internal          default (!defined documentation
+                                     && !defined desc);
+
+    param readable : bool;
+    param readable default configuration != "none";
+    param writable : bool;
+    param writable default configuration != "none";
 
     // required parameter
-    param _attr_type;
+    param _attr_type : const char *;
 }
 
+extern typedef size_t ptrdiff_t;
+
+extern typedef struct {
+    bool valid;
+    bool is_bank;
+    bool is_array;
+    // uint32 array_size;
+    ptrdiff_t port_obj_offset;
+    const char *portname;
+} _attr_parent_obj_proxy_info_t;
```

Comment:
maybe use a longer consistent prefix for internal types, e.g. `_dml_`?

Reply:
Done, this one type was an outlier; every other one type uses `_dml_`.